india_banking/india_banking/doc_events/bank_account/bank_account.py

```
@@ -16,6 +17,13 @@ def validate(doc, method=None):
 	validate_unique_acc_no(doc)
 	update_party_transaction_currency(doc)
 
+	if (
+		(not doc.get_doc_before_save())
+		or doc.has_value_changed("branch_code")
+		and doc.branch_code
```

Ruff (0.15.2)

[warning] Parenthesize `a and b` expressions when chaining `and` and `or` together, to make the precedence clear

Parenthesize the `and` subexpression

(RUF021)